### ISSUE TITLE
Cannot get it to work under Windows :C

### ISSUE THREAD

**Issue by thegoodhen** [duckietown/apriltags3-py#7]
Hello, I love this wrapper! Works perfectly fine for me under Linux. But... I cannot get it to work on my Windows7 machine. 

First I was running into problems such as missing tools, but with the help of windows command line, git bash and 2 different versions of cygwin, I managed to run both the "cmake ." and "make" commands.

Well, the C example which uses opencv failed, since I don't have opencv installed, but whatever.  I just deleted it out of the makefile. The other C example works fine (woohoo!)
Since the os.uname doesn't do anything under Windows, I edited that and just hardcoded the filename to be 'libapriltag.so'. I also edited the relative path, because I have no /lib or /lib64 folders anywhere... I guess those would be created if I typed "make install" ? But that failed with "process_begin: CreateProcess(NULL, ldconfig, ...) failed. make (e=2): System cannot find the file."

But anyway, so I made sure the path to the .so file is correct. It still failed, but I figured out that I was using 32 bit python with 64 bit library. 
So I fixed THAT problem too, but now... It is just randomly failing.
```

  File "L:/personal/tracker/pycharm2/apriltags3.py", line 253, in __init__
    self.tag_detector_ptr = self.libc.apriltag_detector_create()
OSError: exception: access violation writing 0x0000000000000000

  File "L:/personal/tracker/pycharm2/apriltags3.py", line 269, in __init__
    self.libc.apriltag_detector_add_family_bits(self.tag_detector_ptr, self.tag_families['tag36h11'], 2)
OSError: exception: access violation reading 0x000000000000FFF8

L:/personal/tracker/pycharm2/apriltags3.py:462: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.
  parameters = yaml.load(stream)


TESTING WITH A SAMPLE IMAGE

Process finished with exit code -1073740771 (0xC000041D)


  File "L:/personal/tracker/pycharm2/apriltags3.py", line 269, in __init__
    self.libc.apriltag_detector_add_family_bits(self.tag_detector_ptr, self.tag_families['tag36h11'], 2)
OSError: exception: access violation reading 0x000000000000001A

Traceback (most recent call last):
  File "L:/personal/tracker/pycharm2/apriltags3.py", line 459, in <module>
    debug=0)
  File "L:/personal/tracker/pycharm2/apriltags3.py", line 269, in __init__
    self.libc.apriltag_detector_add_family_bits(self.tag_detector_ptr, self.tag_families['tag36h11'], 2)
OSError: exception: access violation reading 0xFFFFFFFFFFFFFFF8




```

Sometimes it doesn't crash, but instead hangs randomly... What is going on... do you know how I could fix it? ;( I ran out of ideas... Thank you!

**Comment by AleksandarPetrov** [comment]
@thegoodhen Unfortunately, we never intended using this on Windows. But it would be great if we make it work. Are you building the library under Windows as well? You  mentioned you are still using an `.so` library, shouldn't that be a `.dll` under Windows?

**Comment by thegoodhen** [comment]
> @thegoodhen Unfortunately, we never intended using this on Windows. But it would be great if we make it work. Are you building the library under Windows as well? You mentioned you are still using an `.so` library, shouldn't that be a `.dll` under Windows?

Oh, I see! That's a shame... Well, worst case scenario is I rewrite my program in C, but I'm not a big fan of doing linear algebra in C, bleh... Also, installing opencv to work with anything else than Python seems like a rather big undertaking.

The .so file appeared after I ran "cmake . " followed by "make"  on my win7 64 bit machine.

I would expect the extension to be .dll as well, but I suppose that's just a matter of the make targets? I am very much a beginner when it comes to this complicated building... I usually just use Visual Studio and create a new project. :D 

Anyway, I tried importing the generated .so file into "DLL EXPORT VIEWER" utility that I downloaded. Keep in mind that I have close to zero idea of what I'm doing and all my attempts are series of educated guesses...  But It did actually list 324 Functions, their addresses and all that. The names appear reasonable. https://i.imgur.com/epucMYK.png

When I opened the .so in Vim, I could find a human-readable text saying "This program cannot be run in dos mode".

Confused by this (it should be a library, not a program, right?), I tried using the dll viewer to import some random .exe files and it showed me the actual functions as well...
Then I tried opening some random .dll files and the text  "This program cannot be run in dos mode" was present in them as well... Well I remember hearing that .dll is essentially just a program without an entry point, but... Hey, I don't know. This is far above my skill level unfortunately. 

Wait, actually, this gives me an idea... Coudln't we use something like rundll to test whether the .so file is compiled well? You know, add an entry point and try it out. Just a guess... Let me know if you have any ideas!

Thank you very much in advance.
